### ISSUE TITLE
The stats ticker does not use the builder's ticker.

### ISSUE THREAD

**Issue by jwiklund** [ben-manes/caffeine#240]
Building a cache loader that uses both stats and expiration (eg recordStats and expireAfterWrite) and providing a ticker for testing builds a cache that uses system time for stats and the provided ticker for expiration.

This can become a problem because the CacheStats constructor validates that variables are not negative, which they can become if the stubbed ticker's time is in the future (or more than System.nanoTime), in general it's possible to test refresh by ensuring that `start time + refresh time` is never more than `System.nanoTime`, but when the computer is restarted (On my linux desktop the jvm seems to use time since startup) tests can not be made to pass until the uptime is more than the refresh time.

**Comment by jwiklund** [comment]
NB I expect this to pass:
```
   class FakeTicker implements Ticker {

    private final AtomicLong nanos = new AtomicLong();

    void advance(Duration duration) {
      nanos.addAndGet(duration.toNanos());
    }

    @Override
    public long read() {
      return nanos.get();
    }
  }

  @Test
  public void test() {
    FakeTicker ticker = new FakeTicker();
    // simulate a low nanoTime by setting the ticker to now.
    ticker.nanos.set(System.nanoTime());

    LoadingCache<Object, String> cache = Caffeine.newBuilder()
        .recordStats()
        .refreshAfterWrite(Duration.ofSeconds(10))
        .ticker(ticker)
        .build((ignoredKey) -> "ok");

    new CacheMetrics(metric_registry).monitor(cache, "country-attributes-cache");

    assertThat(cache.get("1"), is("ok"));

    ticker.advance(Duration.ofSeconds(11));

    assertThat(cache.get("1"), is("ok"));

    cache.stats();
  }
```

Currently it throws 
```
java.lang.IllegalArgumentException
	at com.github.benmanes.caffeine.cache.stats.CacheStats.<init>(CacheStats.java:109)
	at com.github.benmanes.caffeine.cache.stats.ConcurrentStatsCounter.snapshot(ConcurrentStatsCounter.java:95)
	at com.github.benmanes.caffeine.cache.LocalManualCache.stats(LocalManualCache.java:89)
```

**Comment by ben-manes** [comment]
There's a few items on the backlog, but I'll try to wrap them up and release this weekend.

**Comment by ben-manes** [comment]
Released 2.7
